### PR TITLE
Bridges2 support

### PR DESCRIPTION
**Pull request by iparask**
Bridges 2.

PS: Can we reopen the discussion about GPUs and how we define them?

### CODE REVIEW

File: src/radical/saga/adaptors/slurm/slurm_job.py [hunk 1]
```diff
@@ -604,7 +604,14 @@ def _job_run(self, jd):
         # target host specifica
         # FIXME: these should be moved into resource config files
         self._logger.debug ("submit SLURM script to %s", self.rm)
-        if 'bridges' in self.rm.host.lower():
+        if 'bridges2' in self.rm.host.lower():
+             if gpu_count:
+                 # gres resources are specified *per node*
+                 assert(n_nodes), 'need unique number of cores per node'
+
+                 script += "#SBATCH --gres=gpu:v100:8\n"
```
Comment (andre-merzky) [inline src/radical/saga/adaptors/slurm/slurm_job.py:612]: Hmm, shouldn't this trickle down from the `system_architecture` setting in the RP config?

**Review by andre-merzky — COMMENTED** [review]
thanks!

File: src/radical/saga/adaptors/slurm/slurm_job.py (hunk 1 above)
Comment (iparask) [reply src/radical/saga/adaptors/slurm/slurm_job.py:612]: Yes. Let me find how to do it and update

Comment (iparask) [reply src/radical/saga/adaptors/slurm/slurm_job.py:612]: I have a question. RP's config defines `system_architecture` and `gpus_per_node`. RS job description includes the `system_architecture`, but has no means to get the number of GPUs per node. May I suggest the following:

```suggestion
                 script += "#SBATCH --gres=gpu:%s:%s\n" % (system_arch.split(':'))
```

where in the config `system_architecture` is `{"gpu":"v100:8"}.

What do you think?

Comment (mtitov) [reply src/radical/saga/adaptors/slurm/slurm_job.py:612]: btw there is an internal variable that reads `'gpu'` from `system_architecture` -> `gpu_arch`

**Comment by iparask** [comment]
ready!

**Review by mtitov — APPROVED** [review]

**Review by andre-merzky — APPROVED** [review]

**Pull request merged**